File: src/main/java/yuu/rpg/RPG.java
package yuu.rpg;

import org.bukkit.plugin.java.JavaPlugin;


public final class RPG extends JavaPlugin {

    @Override
    public void onEnable() {
        saveDefaultConfig();
        new CustomConfig(this, "UUID.yml").saveDefaultConfig();

        new Join(this);
        new SignJob(this);
        new DeathRespawn(this);
        new JobSkill(this);
        new Skill(this);

        getCommand("job").setExecutor(new JobCommand(this));
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
    }
}

File: src/main/java/yuu/rpg/RPG.kt
package yuu.rpg

import org.bukkit.plugin.java.JavaPlugin
import yuu.rpg.command.JobCommand
import yuu.rpg.command.Spawn
import yuu.rpg.config.CustomConfig
import yuu.rpg.entity.CustomEntities
import yuu.rpg.job.LevelUP
import yuu.rpg.job.SignJob
import yuu.rpg.oputil.OPGUIClick
import yuu.rpg.oputil.OPUtil
import yuu.rpg.other.DeathRespawn
import yuu.rpg.other.Join
import yuu.rpg.skill.Skill
import yuu.rpg.spawnblock.SetSpawnBlock

class RPG : JavaPlugin() {

    override fun onEnable() {

        CustomEntities.registerEntities()
        saveDefaultConfig()
        CustomConfig(this, "UUID.yml").saveDefaultConfig()

        Join(this)
        SignJob(this)
        DeathRespawn(this)
        LevelUP(this)
        Skill(this)
        SetSpawnBlock(this)
        OPGUIClick(this)

        getCommand("job").executor = JobCommand(this)
        getCommand("spawn").executor = Spawn(this)
        getCommand("oputil").executor = OPUtil(this)
    }

    override fun onDisable() {
        CustomEntities.unregisterEntities()
    }
}
